src/fields/DimensionsField.php

```diff
@@ -0,0 +1,199 @@
+<?php
+/**
+ * Dimensions plugin for Craft CMS 3.x
+ *
+ * Replicate the Craft Commerce Dimensions fields as a standalone field type.
+ *
+ * @link      https://www.viget.com/
+ * @copyright Copyright (c) 2020 Trevor Davis
+ */
+
+namespace viget\dimensions\fields;
+
+use Craft;
+use craft\base\ElementInterface;
+use craft\base\Field;
+use yii\db\Schema;
+use craft\helpers\Json;
+use craft\base\PreviewableFieldInterface;
+
+/**
+ * @author    Trevor Davis
+ * @package   Dimensions
+ * @since     1.0.0
+ */
+class DimensionsField extends Field implements PreviewableFieldInterface
+{
+    public $dimensionUnit = 'in';
+    public $weightUnit = 'lb';
+
+    const DIMENSION_UNITS = [
+        'in' => 'Inches',
+        'ft' => 'Feet',
+        'mm' => 'Millimeters',
+        'cm' => 'Centimeters',
+        'm' => 'Meters',
+    ];
+
+    const WEIGHT_UNITS = [
+        'lb' => 'Pounds',
+        'g' => 'Grams',
+        'kg' => 'Kilograms',
+    ];
+
+    /**
+     * @inheritdoc
```

Comment:
what's this do exactly?

Reply:
When you use any sort of documentation thing (like even PHP intellisense) it'll pull all that information from the parent class

![image](https://user-images.githubusercontent.com/199985/78364947-54650800-758c-11ea-8fc3-040d64987122.png)